sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -201,6 +201,16 @@ private[spark] object HiveUtils extends Logging {
     .booleanConf
     .createWithDefault(true)
 
+  val HIVE_THRIFT_SERVER_EXIT_ON_ERROR =
+    buildConf("spark.sql.hive.thriftServer.exitOnThriftCLIServiceError")
```

Comment:
Isn't it a little weird to have a configuration to fix flakiness?

Reply:
Thank you for pointing this out, @dongjoon-hyun. I completely agree with you that creating a new configuration for a test fix would be excessive. However, in this case, we can address both the DevelopAPI and test side issue as a typical use case simultaneously, This way, the solution would not be limited to just fixing the test.